solr/test-framework/src/java/org/apache/solr/cloud/SolrCloudTestCase.java

```diff
@@ -140,28 +143,29 @@ public static void shutdownCluster() throws Exception {
   @BeforeClass
   public static void setPrsDefault() {
     Class<?> target = RandomizedTest.getContext().getTargetClass();
-    if (target != null && target.isAnnotationPresent(NoPrs.class)) return;
-    if (isPRS()) {
-      System.setProperty("solr.prs.default", "true");
+    String existingPrsDefault = EnvUtils.getProperty(PRS_DEFAULT_PROP);
+    String newPrsDefault;
+    if (target != null && target.isAnnotationPresent(NoPrs.class)) {
+      if (existingPrsDefault != null) {
+        System.setProperty(SAVED_PRS_DEFAULT_PROP, existingPrsDefault);
+      }
+      newPrsDefault = "false";
+    } else if (existingPrsDefault != null) {
+      newPrsDefault = existingPrsDefault;
+    } else {
+      newPrsDefault = Boolean.toString(random().nextBoolean());
     }
```

Comment:
Can't we instead rely on TestRuleRestoreSystemProperties, thus don't need to restore whatever the pre-existing sys prop (set at CLI?) is?